cjworkbench/i18n/catalogs/util.py

```diff
@@ -0,0 +1,43 @@
+from babel.messages.pofile import read_po, write_po
+from babel.messages.catalog import Catalog, Message
+from typing import Optional
+
+
+def message_unique_identifier(message: Message):
+    """ Return an immutable element that uniquely identifies a message.
+    Usually, messages are identified by their ID, 
+    but if they have a context it must also be taken into account.
+    """
+    return message.id if not message.context else (message.id, message.context)
+
+
+def find_corresponding_message(catalog: Catalog, message: Message) -> Optional[Message]:
+    """ Search the catalog for a message with the same ID (and context) and return it.
+    """
+    message = catalog.get(message.id, context=message.context)
+    return message if message else None
+
+
+def find_corresponding_string(catalog: Catalog, message: Message) -> Optional[str]:
+    """ Search the catalog for a message with the same ID (and context) and return its string.
+    """
+    corresponding_message = find_corresponding_message(catalog, message)
+    return corresponding_message.string if corresponding_message else None
+
+
+def read_po_catalog(filename: str) -> Optional[Catalog]:
+    """ Try to read a po catalog from the given path.
+    Return None on failure
```

Comment:
Raise an error on failure. None of the callers test for `None` (nor should they). Nix the try/except.